src/response_sender.cc

```
@@ -130,28 +130,30 @@ ResponseSender::Send(
   }
 
   if (has_gpu_output) {
-    AllocatedSharedMemory<char> gpu_buffers_handle =
-        shm_pool_->Load<char>(send_message_payload->gpu_buffers_handle);
-
-    bi::managed_external_buffer::handle_t* gpu_buffers_handle_shm =
-        reinterpret_cast<bi::managed_external_buffer::handle_t*>(
-            gpu_buffers_handle.data_.get() + sizeof(uint64_t));
-    uint64_t* gpu_buffer_count =
-        reinterpret_cast<uint64_t*>(gpu_buffers_handle.data_.get());
-    if (gpu_tensors.size() != *gpu_buffer_count) {
+    AllocatedSharedMemory<GPUBuffersShm> gpu_buffers_handle =
+        shm_pool_->Load<GPUBuffersShm>(
+            send_message_payload->gpu_buffers_handle);
+
+    AllocatedSharedMemory<bi::managed_external_buffer::handle_t>
+        gpu_buffers_handle_shm =
+            shm_pool_->Load<bi::managed_external_buffer::handle_t>(
+                gpu_buffers_handle.data_->buffers);
+    uint64_t gpu_buffer_count = gpu_buffers_handle.data_->buffer_count;
+    if (gpu_tensors.size() != gpu_buffer_count) {
       LOG_INFO
```

Should this be LOG_ERROR or throw instead of LOG_INFO?